aspnet/mvc/controllers/testing.rst

```diff
@@ -1,8 +1,173 @@
-.. include:: /../common/stub-topic.txt
+Testing Controller Logic
+========================
+By `Steve Smith`_
 
-|stub-icon| Testing Controller Logic
-====================================
+Controllers in ASP.NET MVC apps should be small and focused on user-interface concerns. Large controllers that deal with non-UI concerns are more difficult to test and maintain.
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections
+	:local:
+	:depth: 1
+	
+`View sample files <https://github.com/aspnet/Docs/tree/master/aspnet/mvc/controllers/testing/sample>`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/123
+What is Controller Logic
```

Comment:
I don't think you need to cover what a controller is in this article. Instead I would focus on *why* you might want to test controllers in isolation and *how* you can do it. As part of the *how* discussion you could give tips on how to make controller logic easier to test. I would rename this section to something like "Why test controllers".